src/main/scala/com/fulcrumgenomics/bam/LeftAlignIndels.scala

```diff
@@ -0,0 +1,330 @@
+    * mapped. NB: hard-clipped bases are lost. */
+  def apply(rec: SamRecord, target: String): IndelPaddedAlignment = {
+    require(rec.mapped)
+    this.apply(rec.cigar, rec.basesString, target)
+  }
+
+  /** Builds a padded alignment for the given [[Cigar]], query and target target bases. NB: hard-clipped bases are lost. */
+  def apply(cigar: Cigar, query: String, target: String): IndelPaddedAlignment = {
+    require(cigar.lengthOnQuery == query.length,
+      s"query length mismatch: ${cigar.lengthOnQuery} vs. ${query.length}")
+    require(cigar.lengthOnTarget == target.length,
+      s"target length mismatch: ${cigar.lengthOnTarget} vs. ${target.length}")
+
+    val queryBuilder  = new ArrayBuffer[Char]()
+    val targetBuilder = new ArrayBuffer[Char]()
+
+    // Build the padded query
+    cigar.foldLeft(0) { case (index, elem) =>
+      if (elem.lengthOnQuery > 0) { // M, X, EQ, I, S
+        queryBuilder.appendAll(query.slice(index, index + elem.length))
+        index + elem.length
+      }
+      else { // H or D
+        if (elem.operator == CigarOperator.D) {
+          require(elem.operator == CigarOperator.D)
+          queryBuilder.appendAll(IndelString * elem.length)
+        }
+        index
+      }
+    }
+
+    // Build the padded target
+    cigar.foldLeft(0) { case (index, elem) =>
+      // target
+      if (elem.lengthOnTarget == 0) { // S, H, I
+        if (elem.operator == CigarOperator.I) targetBuilder.appendAll(IndelString * elem.length)
+        else if (elem.operator == CigarOperator.S) targetBuilder.appendAll(SoftClipString * elem.length)
+      }
+      else { // M, X, EQ, D
+        targetBuilder.appendAll(target.slice(index, index + elem.length))
+      }
+      index + elem.lengthOnTarget
+    }
+
+    new IndelPaddedAlignment(queryBuilder.mkString, targetBuilder.mkString)
+  }}
+
+/** Stores an alignment padded for indels.
+  *
+  * The query and target strings are padded as follows:
+  * - an inserted base (in the query) has a corresponding gap character ('-') in the target
+  * - a deleted base (in the target) has a corresponding gap character ('-') in the query
+  * - a soft-clipped base (in the query) has a corresponding soft-clip character ('X') in the target
+  *
+  * For example:
+  *
+  * query:  AAAGATTACAGATT--ACA
+  * target: XXXGA--ACAGATTAAACA
+  *
+  * In this example, query bases 1-3 ("AAA") are soft-clipped, query bases 6-7 ("TT") are inserted, and target bases
+  * 10-11 ("AA") are deleted.
+  *
+  * */
+case class IndelPaddedAlignment private(query: String, target: String){
+  import IndelPaddedAlignment.{Indel, SoftClip}
+  require(query.length == target.length,
+    f"Lengths differ ${query.length} != ${target.length}\n${query.mkString("")}\n${target.mkString("")}")
+  require(length > 0)
+
+  def length: Int = query.length
+
+  def cigar(useEqualsAndX: Boolean = false): Cigar = {
```

Comment:
note to self, this could be used to create an `Alignment` object from padded strings

Reply:
I think we should offer a method to build an `Alignment` object from padded strings :)